src/mechanics/mesh/UnitMeshFem.cpp

```diff
@@ -54,10 +54,11 @@ MeshFem UnitMeshFem::CreateQuads(int numX, int numY)
 
 MeshFem UnitMeshFem::Transform(MeshFem&& oldMesh, std::function<Eigen::VectorXd(Eigen::VectorXd)> f)
 {
+    MeshFem newMesh = std::move(oldMesh);
     // Build a group to avoid duplicates. Otherwise, the transformation is applied multiple times.
-    // This is, however, a bit of a bottleneck for big meshes. 
-    for (auto& node : oldMesh.NodesTotal())
+    // This is, however, a bit of a bottleneck for big meshes.
+    for (auto& node : newMesh.NodesTotal())
         node.SetValues(f(node.GetValues()));
 
-    return oldMesh;
+    return newMesh;
```

Comment:
I'd rather have `return std::move(oldMesh);` here instead of the extra line at the beginning. And the comment is not applicable anymore.